codec/src/main/java/io/netty/handler/codec/DefaultHeaders.java

```diff
@@ -1013,7 +1013,11 @@ protected void validateName(NameValidator<K> validator, boolean forAdd, K name)
     }
 
     protected void validateValue(ValueValidator<V> validator, K name, V value) {
-        validator.validate(value);
+        try {
+            validator.validate(value);
+        } catch (IllegalArgumentException e) {
+            throw new IllegalArgumentException("Validation failed for header '" + name + "' : " + e.getMessage(), e);
```

Comment:
Throwing a stack trace is nothing less than making applications vulnerable to Denial-of-Service (DoS). This can be abused easily.

Consider making it stackless.

Reply:
@normanmaurer WDYT?

Reply:
I am not inclined to make this exception stackless, at the very most I can maybe remove the cause, but making it stackless severely hampers debugging ability and is not even the standard norm in the Netty codebase.

> Throwing a stack trace is nothing less than making applications vulnerable to Denial-of-Service (DoS)

If someone is trying to DoS an application by sending invalid headers, I don't think generating an exception stacktrace would be the bottleneck. Logging the stack trace might be but there are other mitigations for that in the logger configurations which applications need to use anyways.